zebra-instructions.md

```diff
@@ -74,18 +77,72 @@ These fixes disable mining pool operator payments and miner payments: they just
     [mining]
     miner_address = 't27eWDgjFYJGVXmzrXeVjnb5J3uXDM9xH9v'
     ```
+
     </details>
+
 3. [Run Zebra](https://zebra.zfnd.org/user/run.html) with the `getblocktemplate-rpcs` feature:
+
     ```sh
     cargo run --release --features "getblocktemplate-rpcs"
     ```
+
 4. Wait a few hours for Zebra to sync to the testnet tip (on mainnet this takes 2-3 days)
 
-## Install and run a mining pool
+## Install `s-nomp`
+
+### Arch-specific instructions
+
+#### Install dependencies:
+
+1. Install [`redis`](https://redis.io/docs/getting-started/) and run it on the default port:
+
+    ```sh
+    sudo pacman -S redis
+    sudo systemctl start redis
+    ```
+
+2. Install and activate [`nvm`](https://github.com/nvm-sh/nvm#installing-and-updating):
+
+    ```sh
+    sudo pacman -S nvm
+    unset npm_config_prefix
+    source /usr/share/nvm/init-nvm.sh
+    ```
+
+3. Install `boost` and `libsodium` development libraries:
+
+    ```sh
+    sudo pacman -S boost libsodium
+    ```
+
+#### Install `s-nomp`:
+
+1. `git clone https://github.com/ZcashFoundation/s-nomp && cd s-nomp`
+
+2. Use the Zebra configs: `git checkout zebra-mining`
+
+    - TODO: change the fork in `package.json` to ZcashFoundation
+
+3. Use node 8.11.0:
+
+    ```sh
+    nvm install 8.11.0
+    nvm use 8.11.0
```

Comment:
I think we should switch all the instructions to use node 8.11, because we don't know which other platforms 8.17 will fail on.

What was the error message?

Reply:
Here's the output of running `npm start`:

```
> LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$PWD/node_modules/stratum-pool/node_modules/equihashverify/build/Release/:$PWD/node_modules/equihashverify/build/Release/ node init.js

2023-02-21 01:08:12 [POSIX]     [Connection Limit] (Safe to ignore) POSIX module not installed and resource (connection) limit was not raised
2023-02-21 01:08:12 [Master]    [zcash_testnet.json] Pool has same configured coin file coins/testnet/zec.json as zcash.json pool
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! s-nomp@0.0.9 start: `LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$PWD/node_modules/stratum-pool/node_modules/equihashverify/build/Release/:$PWD/node_modules/equihashverify/build/Release/ node init.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the s-nomp@0.0.9 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

I went with 8.11 because that's what the original S-nomp repo recommends. I was actually wondering why you used 8.17.

Reply:
8.17 is the latest release in the latest compatible release series, so it includes all the available security fixes.

Reply:
Here's the error. The first line is my prompt.
```
: s-nomp: update-mining-setup-instructions; npm start

> s-nomp@0.0.9 start /home/m/zcash/teor/s-nomp
> LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$PWD/node_modules/stratum-pool/node_modules/equihashverify/build/Release/:$PWD/node_modules/equihashverify/build/Release/ node init.js

/home/m/zcash/teor/s-nomp/node_modules/bindings/bindings.js:135
  throw err;
  ^

Error: Could not locate the bindings file. Tried:
 → /home/m/zcash/teor/s-nomp/node_modules/bignum/build/bignum.node
 → /home/m/zcash/teor/s-nomp/node_modules/bignum/build/Debug/bignum.node
 → /home/m/zcash/teor/s-nomp/node_modules/bignum/build/Release/bignum.node
 → /home/m/zcash/teor/s-nomp/node_modules/bignum/out/Debug/bignum.node
 → /home/m/zcash/teor/s-nomp/node_modules/bignum/Debug/bignum.node
 → /home/m/zcash/teor/s-nomp/node_modules/bignum/out/Release/bignum.node
 → /home/m/zcash/teor/s-nomp/node_modules/bignum/Release/bignum.node
 → /home/m/zcash/teor/s-nomp/node_modules/bignum/build/default/bignum.node
 → /home/m/zcash/teor/s-nomp/node_modules/bignum/compiled/8.17.0/linux/x64/bignum.node
 → /home/m/zcash/teor/s-nomp/node_modules/bignum/addon-build/release/install-root/bignum.node
 → /home/m/zcash/teor/s-nomp/node_modules/bignum/addon-build/debug/install-root/bignum.node
 → /home/m/zcash/teor/s-nomp/node_modules/bignum/addon-build/default/install-root/bignum.node
 → /home/m/zcash/teor/s-nomp/node_modules/bignum/lib/binding/node-v57-linux-x64/bignum.node
    at bindings (/home/m/zcash/teor/s-nomp/node_modules/bindings/bindings.js:126:9)
    at Object.<anonymous> (/home/m/zcash/teor/s-nomp/node_modules/bignum/index.js:1:92)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/home/m/zcash/teor/s-nomp/node_modules/base58-native/base58.js:2:14)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! s-nomp@0.0.9 start: `LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$PWD/node_modules/stratum-pool/node_modules/equihashverify/build/Release/:$PWD/node_modules/equihashverify/build/Release/ node init.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the s-nomp@0.0.9 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

Reply:
I quickly tried it again, and I got a different error:

```
> s-nomp@0.0.9 start /home/m/zcash/teor/s-nomp
> LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$PWD/node_modules/stratum-pool/node_modules/equihashverify/build/Release/:$PWD/node_modules/equihashverify/build/Release/ node init.js

/home/m/zcash/teor/s-nomp/node_modules/bindings/bindings.js:135
  throw err;
  ^

Error: Could not locate the bindings file. Tried:
 → /home/m/zcash/teor/s-nomp/node_modules/equihashverify/build/equihashverify.node
 → /home/m/zcash/teor/s-nomp/node_modules/equihashverify/build/Debug/equihashverify.node
 → /home/m/zcash/teor/s-nomp/node_modules/equihashverify/build/Release/equihashverify.node
 → /home/m/zcash/teor/s-nomp/node_modules/equihashverify/out/Debug/equihashverify.node
 → /home/m/zcash/teor/s-nomp/node_modules/equihashverify/Debug/equihashverify.node
 → /home/m/zcash/teor/s-nomp/node_modules/equihashverify/out/Release/equihashverify.node
 → /home/m/zcash/teor/s-nomp/node_modules/equihashverify/Release/equihashverify.node
 → /home/m/zcash/teor/s-nomp/node_modules/equihashverify/build/default/equihashverify.node
 → /home/m/zcash/teor/s-nomp/node_modules/equihashverify/compiled/8.17.0/linux/x64/equihashverify.node
 → /home/m/zcash/teor/s-nomp/node_modules/equihashverify/addon-build/release/install-root/equihashverify.node
 → /home/m/zcash/teor/s-nomp/node_modules/equihashverify/addon-build/debug/install-root/equihashverify.node
 → /home/m/zcash/teor/s-nomp/node_modules/equihashverify/addon-build/default/install-root/equihashverify.node
 → /home/m/zcash/teor/s-nomp/node_modules/equihashverify/lib/binding/node-v57-linux-x64/equihashverify.node
    at bindings (/home/m/zcash/teor/s-nomp/node_modules/bindings/bindings.js:126:9)
    at Object.<anonymous> (/home/m/zcash/teor/s-nomp/node_modules/equihashverify/index.js:1:99)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/home/m/zcash/teor/s-nomp/node_modules/stratum-pool/lib/algoProperties.js:1:72)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! s-nomp@0.0.9 start: `LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$PWD/node_modules/stratum-pool/node_modules/equihashverify/build/Release/:$PWD/node_modules/equihashverify/build/Release/ node init.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the s-nomp@0.0.9 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

Reply:
In order to get rid of the error, I have to 

```sh
nvm use 8.11.0
npm update
npm install
```

and then

```sh
npm rebuild
```

and after that I can

```sh
npm start
```

Reply:
> I think we should switch all the instructions to use node 8.11, because we don't know which other platforms 8.17 will fail on.

Done in [3acdf17](https://github.com/ZcashFoundation/s-nomp/pull/9/commits/3acdf17600e37c12ec10ab85c93245e9f7b7eacf)